chemprop/features/utils.py

```diff
@@ -76,6 +77,8 @@ def load_valid_atom_or_bond_features(path: str, smiles: List[str]) -> List[np.nd
     if extension == '.npz':
         container = np.load(path)
         features = [container[key] for key in container]
+        if len(features) != len(smiles):
+            raise ValueError(f'Different number of valid smiles than features in {path}. Features saved in npz files cannot be filtered to include only valid molecules. When using npz, all molecule datapoints must be valid and in the order corresponding to the features file.')
```

Comment:
Is this a `ValueError`? It seems a little more specific than that and could potentially raise problems for `try/except` blocks in client code as a result of its generality. I.e., `path` could raise a `ValueError` earlier in the code (e.g., L75/78). If a client wants to catch this `ValueError` (on L81), they'll also catch that one mistakenly. All of this is an edge case really, but something to think about in terms of adding more specific exceptions.

Reply:
Seems a reasonable improvement in the code practice. What would you suggest? RunTimeError? Or making a custom error class for this case?

Reply:
probably a custom exception inside a file `chemrop/exceptions.py`. You can name it whatever you want as long as its descriptive- maybe `InvalidFeaturesError`?

Reply:
Good suggestion

Reply:
Coming back around to this with some second thoughts. ValueError might be general but I don't think it's wrong or in conflict with other errors here. If you supply a bad path, L75/78 will supply other error types (TypeError or FileNotFoundError), so I don't know that I see a place where a try/except would fail.

More generally, Chemprop is filled with ValueErrors that are similar. We should find clusters of these so that they can follow the same few custom error classes, rather than adding narrowly tailored ones piecemeal. This would be better handled as a separate PR.

Reply:
You can supply a valid filepath that's either (i) invalid format (i.e., not able to parsed an `.npy` file or (ii) a valid `.npy` file that is improper for use in *this* function. There's the possible client code where a user `try`s to pass in a file and `excepts` a `ValueError` for an invalid `.npy` file (reason i), but their problem is actually that the `.npy` is of the incorrect dimension. Obviously, the function should `raise` an `Exception` in either case, but now their very specific bug is being silently handled and not being communicated properly (because they've caught a different `ValueError` than they think)

Using `ValueError` elsewhere in the library doesn't sound too compelling a reason to me to avoid using a specific `*Error` here. I agree, however, that this might be better served as a separate PR

Reply:
Ah! The parsing errors within np had escaped me.